builtin/providers/powerdns/client.go

```diff
@@ -111,6 +124,26 @@ func parseId(recId string) (string, string, error) {
 	}
 }
 
+// Detects the API version in use on the server
```

Comment:
This piece of code is breaking my .tf deployment as it tries to detect the API version before we launch the instance where PDNS is going to be hosted: here is my provider's config and you can see that my powerdns_server_url link is a place holder of a resource that will be created:

```
# Power DNS Provider
provider "powerdns" {
  api_key    = "${var.pdns_api_key}"
  #server_url = "http://${openstack_compute_instance_v2.admin_node.floating_ip}:8081"
}
```
obviously my terraform plan is failing with:
Error refreshing state: 1 error(s) occurred:
```
* Error setting up PowerDNS client: Error during parsing request URL: parse http://${openstack_compute_instance_v2.admin_node.floating_ip}:8081/api/v1/servers: invalid character "{" in host name
```
Wouldn't be better to check the server API version prior creating the records?